samcli/commands/init/interactive_init_flow.py

```diff
@@ -338,16 +339,45 @@ def get_sorted_runtimes(runtime_option_list):
         sorted list of possible runtime to be selected
     """
     runtime_list = []
-    for runtime in runtime_option_list:
+    supported_runtime_list = get_supported_runtime(runtime_option_list)
+    for runtime in supported_runtime_list:
         extractLanguageFromRuntime = re.split(r"\d", runtime)[0]
         extractVersionFromRuntime = re.search(r"\d.*", runtime).group()
         runtime_list.append((extractLanguageFromRuntime, extractVersionFromRuntime))
 
-    runtime_list = sorted(runtime_option_list, key=functools.cmp_to_key(compare_runtimes))
+    runtime_list = sorted(supported_runtime_list, key=functools.cmp_to_key(compare_runtimes))
     sorted_runtime = ["".join(runtime_tuple) for runtime_tuple in runtime_list]
     return sorted_runtime
 
 
+def get_supported_runtime(runtime_list):
+    """
+    Returns a list of only runtimes supported by the current version of SAMCLI.
+    This is the list that is presented to the customer to select from.
+
+    Parameters
+    ----------
+    runtime_list : list
+        List of runtime
+
+    Returns
+    -------
+    list
+        List of supported runtime
+    """
+    supported_runtime_list = []
+    error_message = ""
+    for runtime in runtime_list:
+        if runtime not in INIT_RUNTIMES:
+            if not error_message:
+                error_message = "Additional runtimes may be available in the latest SAM CLI version. \
+                    Upgrade your SAM CLI to see the full list."
+                LOG.debug(error_message)
+            continue
+        supported_runtime_list.append(runtime)
+    return supported_runtime_list
```

Comment:
Shorter way of doing this
```suggestion
    supported_runtime_list = list(set(runtime) & set(INIT_RUNTIMES))
    if len(supported_runtime_list) != len(runtime):
        LOG.debug(
                    "Additional runtimes may be available in the latest SAM CLI version. Upgrade your SAM CLI to see the full list."
                )
    return supported_runtime_list
```

Reply:
This doesn't covers an edge where we INIT_runtime doesn't all the runtime, ie custom runtime support.

Reply:
but if INIT_runtime doesn't contain all the runtimes, it would throw an error anyway because of the check `if runtime not in INIT_RUNTIMES`, no? (I believe the `supported_runtime_list.append(runtime)` wouldn't be run because of the `continue` before it). 
Also, if that's a valid edge case, could we add a test to cover it?

Reply:
if you need to return all of the runtimes in `runtime_list` then you could just update my suggestion to return `runtime_list` instead of `supported_runtime_list`. But I'm not sure if I understand correctly